src/common/utils/prompt-utils.ts

```diff
@@ -39,6 +39,9 @@ export default class PromptUtils {
       }
       for (const prompt of prompts) {
         if ((prompt.when && prompt.default === undefined) || prompt.when === undefined) {
+          if (prompt.name === 'deploy' && prompt.type === 'confirm') {
```

Comment:
There are a few other places within the CLI where we use `confirm` prompts, so ideally this would be a solution that would cover current and future cases/commands. Let me know if you'd like me to have a look too and we can see what we can come up with

Reply:
@mtgoncalves1 TJ had the idea of printing the error in the `when` for each necessary prompt. For example

```ts
const confirmation = await inquirer.prompt({
      type: 'confirm',
      name: 'deploy',
      message: 'Would you like to apply?',
      when: () => {
        if (isCI) {
          this.error('--auto-approve is required in ci pipelines');
        }
        return true;
      },
    });
```

What are your thoughts on that? If we're going to use custom error messages per prompt, the error should just be part of it rather than in the util file

Reply:
I like this since it is quite hard to figure out where the error is coming from.